### ISSUE TITLE
ClassCastException

### ISSUE THREAD

**Issue by markwoon** [vsch/idea-multimarkdown#64]
Getting lots of:  

```
com.vladsch.idea.multimarkdown.psi.impl.MultiMarkdownExplicitLinkImpl cannot be cast to com.vladsch.idea.multimarkdown.psi.MultiMarkdownWikiLink
java.lang.ClassCastException: com.vladsch.idea.multimarkdown.psi.impl.MultiMarkdownExplicitLinkImpl cannot be cast to com.vladsch.idea.multimarkdown.psi.MultiMarkdownWikiLink
	at com.vladsch.idea.multimarkdown.annotator.MultiMarkdownAnnotator.addChangeWikiLinkToExplicitLink(MultiMarkdownAnnotator.java:77)
	at com.vladsch.idea.multimarkdown.annotator.MultiMarkdownAnnotator.annotateLinkRef(MultiMarkdownAnnotator.java:235)
	at com.vladsch.idea.multimarkdown.annotator.MultiMarkdownAnnotator.addChangeExplicitLinkToWikiLink(MultiMarkdownAnnotator.java:131)
	at com.vladsch.idea.multimarkdown.annotator.MultiMarkdownAnnotator.annotate(MultiMarkdownAnnotator.java:68)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.a(DefaultHighlightVisitor.java:162)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:103)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:354)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$100(GeneralHighlightingPass.java:65)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$3.run(GeneralHighlightingPass.java:283)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:308)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$200(GeneralHighlightingPass.java:65)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$4.run(GeneralHighlightingPass.java:314)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:87)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:311)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:280)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:219)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:82)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:67)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:444)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1169)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:435)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:432)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:206)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)

```

**Comment by vsch** [comment]
@markwoon, I have this fixed but not released yet. I can upload a fix to this repository so you can download and install it. It would also have a few other things fixed. Let me know if you want the pre-release fix.

**Comment by markwoon** [comment]
I can wait for now.  You're on a biweekly schedule?

**Comment by vsch** [comment]
Yes, but there are a few bugs that made it into the release that I want fixed so I will make a release in a release in a few days.

**Comment by vsch** [comment]
a few pre-release fixes https://github.com/vsch/idea-multimarkdown/raw/master/dist/idea-multimarkdown.1.2.1.1.zip

### 1.2.1.1 - Bug Fixes and Enhancements

#### Basic Edition
- Fix #62: Plugin exhausted thread resources if a lot of projects were open 
- Add annotation if wiki link ends in .md, with quick fix to remove .md extension  
- Fix: buy license button in settings navigated to the get trial license page instead of the buy license page.
- Fix #60: use another class for string to input stream conversion that is universally support across os types.
- Fix #63: Unresolved links were not always highlighted as error
- Add a change to quick fix on an unresolved link will apply the same change to all links that match the link being modified. 

#### Enhanced Edition

- Fix #64: Explicit link annotation had WikiLink cast causing CastClassException.
